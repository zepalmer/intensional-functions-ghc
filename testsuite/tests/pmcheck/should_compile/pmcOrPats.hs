{-# OPTIONS_GHC -Wincomplete-patterns #-}
{-# LANGUAGE OrPatterns #-}

data T = A | B
data U = V | W

g :: T -> U -> Int
g (one of A,B) V = 0
g B (one of V,W) = 1

h A (one of _,W) B = 0
h B (one of V,_) B = 1
h (one of A,B) _ B = 2

z (one of 1,2,1) = 0
z (one of 3,2) = 1
z 1 = 2

main = print 2