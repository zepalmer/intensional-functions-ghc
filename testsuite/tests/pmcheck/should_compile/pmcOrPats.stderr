
pmcOrPats.hs:8:1: warning: [GHC-62161] [-Wincomplete-patterns (in -Wextra)]
    Pattern match(es) are non-exhaustive
    In an equation for ‘g’: Patterns of type ‘T’, ‘U’ not matched: A W

pmcOrPats.hs:11:1: warning: [GHC-62161] [-Wincomplete-patterns (in -Wextra)]
    Pattern match(es) are non-exhaustive
    In an equation for ‘h’:
        Patterns of type ‘T’, ‘U’, ‘T’ not matched:
            A _ A
            B V A
            B W A

pmcOrPats.hs:13:1: warning: [GHC-53633] [-Woverlapping-patterns (in -Wdefault)]
    Pattern match is redundant
    In an equation for ‘h’: h (one of A, B) _ B = ...

pmcOrPats.hs:15:1: warning: [GHC-62161] [-Wincomplete-patterns (in -Wextra)]
    Pattern match(es) are non-exhaustive
    In an equation for ‘z’:
        Patterns of type ‘a’ not matched: p where p is not one of {3, 1, 2}

pmcOrPats.hs:17:1: warning: [GHC-53633] [-Woverlapping-patterns (in -Wdefault)]
    Pattern match is redundant
    In an equation for ‘z’: z 1 = ...
